<?xml version="1.0"?>
<launch>
  <arg name="script" default="nav_cloning_node_pytorch_plot1_willow.py"/>
  <arg name="script2" default="nav_cloning_node_pytorch_plot2_willow.py"/>


  <!-- <arg name="mode" default="use_dl_output"/> -->
  <arg name="mode" default="change_dataset_balance"/>
  <!-- <arg name="mode" default="follow_line"/> -->



  <!-- <willow_my_config> -->
  <arg name="robot_x" default="2.915899" />
  <arg name="robot_y" default="0.717698" />
  <arg name="robot_Y" default="0.024363" />

  <!-- 2-3 -->
  <!-- <arg name="robot_x" default="-10.1856145815951" /> -->
  <!-- <arg name="robot_y" default="28.441895426996574" /> -->
  <!-- <arg name="robot_Y" default="-1.57" /> -->


  <!-- 2-3_real -->
  <!-- <arg name="robot_x" default="-1.480409" /> -->
  <!-- <arg name="robot_y" default="-3.962090" /> -->
  <!-- <arg name="robot_z" default="0.116067" /> -->
  <!-- <arg name="robot_Y" default="-0.024075" /> -->

  
  
  <rosparam param="/nav_cloning_node/mode" subst_value="True">$(arg mode)</rosparam>

  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find nav_cloning)/world/willow_garage.world"/> 
    <!-- <arg name="world_name" value="$(find real_tsudanuma2-3_sim)/world/tsudanuma_scan.world"/>  -->
    <!-- <arg name="world_name" value="$(find nav_cloning)/world/Tsudanuma_2-3.world"/>  -->
    <arg name="paused" value="false"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="false"/>
    <arg name="headless" value="false"/>
    <arg name="debug" value="false"/>
  </include>

  <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_waffle_pi.urdf.xacro" />
  <node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf" args="-urdf -model mobile_base -x $(arg robot_x) -y $(arg robot_y) -z 0.0 -Y $(arg robot_Y) -param robot_description" />
  <!-- <node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf" args="-urdf -model mobile_base -x $(arg robot_x) -y $(arg robot_y) -z $(arg robot_z)  -Y $(arg robot_Y) -param robot_description" /> -->


  <node pkg="timed_roslaunch" type="timed_roslaunch.sh"
        args="5 nav_cloning nav_cloning.launch script:=$(arg script)"
        name="timed_roslaunch" output="screen" />

  <node pkg="timed_roslaunch" type="timed_roslaunch.sh"
        args="7 nav_cloning nav_cloning2.launch script2:=$(arg script2)"
        name="timed_roslaunch2" output="screen" />

  <!-- <node pkg="timed_roslaunch" type="timed_roslaunch.sh"
        args="10 nav_cloning turtlebot3_navigation_manager.launch model:=waffle map_file:=$(find nav_cloning)/maps/cit_3f_map.yaml"
        name="timed_roslaunch3" >
  </node> -->

  <node pkg="timed_roslaunch" type="timed_roslaunch.sh"
        args="10 nav_cloning turtlebot3_navigation_manager.launch model:=waffle map_file:=$(find nav_cloning)/maps/willowgarage.yaml"
        name="timed_roslaunch3" >
  </node>
  
  <node pkg="timed_roslaunch" type="timed_roslaunch.sh"
        args="20 nav_cloning start_manager_nav.launch"
        name="timed_roslaunch4" >
  </node>

</launch>
